documentation/reference/API.md

```diff
@@ -0,0 +1,136 @@
+## Ockam API workers
+
+Ockam Workers serve as actors being able to handle and send Ockam Messages.
+Ockam Routing protocol only describes unidirectional messages with very simple
+message structure. Internal structure of the message payload is up to the worker
+implementation.
+
+A common messaging pattern, especially with the online services is request-response,
+which can be implemented with Ockam Messages.
+
+API workers provide a relatively simple enough framework for workers
+to implement request-response messaging pattern.
+
+API worker communication consists of the API worker and the Client worker exchanging
+Request and Response messages.
+
+API worker receives Requests and replies with Responses.
+
+Client worker sends a Request and waits for a Response.
+
+
+### Request and Response format
+
+Request and response payloads consist of a CBOR encoded header and a binary body.
+
+Headers are described in the following CDDL schema:
+
+```
+request_header = {
+    ?0: 7586022,
+     1: id,
+     2: path,
+     3: method,
+     4: has_body
+}
+
+response_header = {
+    ?0: 9750358,
+     1: id,
+     2: request_id,
+     3: status,
+     4: has_body
+}
+
+error = {
+    ?0: 5359172,
+    ?1: path,
+    ?2: method,
+    ?3: message
+}
+
+id = uint
+request_id = uint
+path = text ;; additional metadata about the request
+has_body = bool ;; if true - header is followed by the body
+
+method = 0 ;; GET
+       / 1 ;; POST
+       / 2 ;; PUT
+       / 3 ;; DELETE
+       / 4 ;; PATCH
+
+status = uint
+;; 200 OK
+;; 400 Bad request
+;; 401 Unauthorized
+;; 404 Not found
+;; 405 Method not allowed
+;; 409 Resource exists
+;; 500 Internal server error
+;; 501 Not implemented
+```
+
+The format mimics HTTP request/response structures.
+
+### API request/response reference format
+
+To describe an API, we can use the following format:
+
+````
+### Cats API
+
+Worker address: "cats"
+
+#### See the cats face
+Method: GET
+Path: ":cat_name/face"
+Request: ""
```

Comment:
maybe Request-Body  or explain that it is a second struct following the request header?

Reply:
I tried to do that, but it looked cumbersome to write. We can change that if we see that it's confusing.